modin/core/dataframe/base/dataframe/dataframe.py

```diff
@@ -0,0 +1,541 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module contains class ModinDataframe.
+
+ModinDataframe is a parent abstract class for any dataframe class.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Hashable, Optional, Callable, Union, Dict
+from enum import Enum
+
+
+class Axis(Enum):  # noqa: PR01
+    """
+    An enum that represents the axis argument provided to the algebra operators.
+
+    The enum has 3 values - ROW_WISE to represent the row axis, COL_WISE to represent the
+    column axis, and CELL_WISE to represent no axis. ROW_WISE operations iterate over the rows
+    COL_WISE operations over the columns, and CELL_WISE operations over any partitioning scheme.
```

Comment:
CELL_WISE operations can also occur over block partitions of the DataFrame, which is why I didn't want to specify cell-by-cell. I can change it if it would be clearer otherwise? @devin-petersohn

Reply:
If cell-wise partitioning is unclear to someone that it means cell-wise/block-wise partitioning, we can change the phrase from
```
and CELL_WISE operations over any partitioning scheme.
```
to
```
and CELL_WISE operations over cell-wise (block-wise) partitioning scheme.
```

Reply:
fix: replace `any partitioning scheme` -> `over row-wise, column-wise, or block-wise partitioning schemes`.